projects/angular-cld/src/lib/cloudinary-sdk-analytics.spec.ts

```diff
@@ -0,0 +1,217 @@
+import { Component, DebugElement } from '@angular/core';
+import {ComponentFixture, fakeAsync, TestBed, tick} from '@angular/core/testing';
+import { By } from '@angular/platform-browser';
+import { Cloudinary } from './cloudinary.service';
+import CloudinaryConfiguration from './cloudinary-configuration.class';
+import { CloudinaryImage } from './cloudinary-image.component';
+import { CloudinaryTransformationDirective } from './cloudinary-transformation.directive';
+import {LazyLoadDirective } from './cloudinary-lazy-load.directive';
+import { CloudinaryPlaceHolder } from'./cloudinary-placeholder.component';
+
+describe('Tests for sdk versionID on image tag', () => {
+  describe('Config with urlAnalytics not set', () => {
+    let localCloudinary: Cloudinary = new Cloudinary(require('cloudinary-core'),
+      { cloud_name: '@@fake_angular_sdk@@' } as CloudinaryConfiguration);
+
+    beforeEach(() => {
+      spyOn(localCloudinary, 'toCloudinaryAttributes').and.callThrough();
+      spyOn(localCloudinary, 'url').and.callThrough();
+      spyOn(localCloudinary, 'responsive').and.callThrough();
+    });
+    @Component({
+      template: `<cl-image responsive public-id="sample"></cl-image>`
+    })
+    class TestComponent { }
+
+    let fixture: ComponentFixture<TestComponent>;
+    let des: DebugElement;  // the elements w/ the directive
+
+    beforeEach(() => {
+      fixture = TestBed.configureTestingModule({
+        declarations: [CloudinaryImage, TestComponent],
+        providers: [{ provide: Cloudinary, useValue: localCloudinary }]
+      }).createComponent(TestComponent);
+
+      fixture.detectChanges(); // initial binding
+      expect(localCloudinary.responsive).toHaveBeenCalled();
+
+      // all elements with an attached CloudinaryImage
+      des = fixture.debugElement.query(By.directive(CloudinaryImage));
+    });
+
+    it('creates an img without encoding', () => {
+      const img = des.children[0].nativeElement as HTMLImageElement;
+      expect(img.attributes.getNamedItem('src').value).toEqual('http://res.cloudinary.com/@@fake_angular_sdk@@/image/upload/sample');
+    });
+  });
+
+  describe('Config with urlAnalytics set to true', () => {
+    let localCloudinary: Cloudinary = new Cloudinary(require('cloudinary-core'),
+      { cloud_name: '@@fake_angular_sdk@@', urlAnalytics: true} as CloudinaryConfiguration);
```

Comment:
Is the config name called `urlAnalytics` in all SDKs?

Reply:
yes it should since the video player has reserved analytics in the options. We ise urlAnalytics in Core, Node, and Angular

Reply:
Please make sure the spec is updated and that other SDKs will call it the same

Reply:
See this please- https://cloudinary.atlassian.net/wiki/spaces/RDKB/pages/742195511/Angular+analytics+implementation+details